docs/public/deploying-a-devfile-using-odo.adoc

```diff
@@ -29,29 +29,50 @@ An example deployment scenario:
 
 == Deploying your first devfile
 
-===== Prerequisites for a Kubernetes Cluster
+=== Ingress Setup 
 
-* Before proceeding, you must know your ingress domain name or ingress IP to specify `--host` for `odo url create`.
-+
-Ingress IP is usually the external IP of ingress controller service. For example, for Minikube running in a virtual machine you can get it by
-`minikube ip`. Checkout this https://kubernetes.io/docs/concepts/services-networking/ingress/[document] to know more about ingress.
+You will need to provide an ingress domain name for the services you create with odo, which you will specify via the `--host` argument with `odo url create`.
 
-===== Prerequisites for an OpenShift Cluster
+An easy way to do this is to use the https://nip.io/[nip.io] service to create host names mapping to the external IP of your ingress controller service. 
 
-* Create a project to keep your source code, tests, and libraries organized in a separate single unit.
-* You must also know your ingress domain name or ingress IP to specify `--host` for `odo url create`. This can be found using `crc ip`.
-+
-Ingress IP is usually the external IP of ingress controller service. For example, for link:https://developers.redhat.com/products/codeready-containers/overview[CodeReady Containers] running in a virtual machine you can get it by
-`crc ip`. Checkout this https://kubernetes.io/docs/concepts/services-networking/ingress/[document] to know more about ingress.
+In the commands below we assume you are using the https://nip.io/[nip.io] service and https://minikube.sigs.k8s.io/docs/[minikube], e.g.: 
+
+[source,sh]
+----
+  $ odo url create --host $(minikube ip).nip.io
+----
+
+===== Minikube Cluster Ingress Setup
+
+Enable the ingress addon in minikube. 
+
+[source,sh]
+----
+  $ minikube addons enable ingress
+----
+
+With Minikube running in a virtual machine, the ingress controller IP address is obtained via `minikube ip` (the value is `192.168.99.100` in the sample output shown below).
+
+===== OpenShift Cluster Ingress Setup
 
-. Log in to an OpenShift cluster:
+For link:https://developers.redhat.com/products/codeready-containers/overview[CodeReady Containers] running in a virtual machine, for example, you can get it by `crc ip`.
+Note you might not need to use the ingress in OpenShift, where you can use routes instead.
+
+===== Ingress Notes
+
+Of course there are other options but this approach avoids the need to edit the `/etc/hosts/` file with each created service URL.
+
+Checkout this https://kubernetes.io/docs/concepts/services-networking/ingress/[document] to know more about ingress.
+
+=== First Steps
+
+* Login to your cluster (unnecessary if you've used other standard methods, e.g. kubectl to establish the current context)
```

Comment:
`s/unnecessary/not required/`. Suggesting this because "unnecessary" gives quite different perspective to things, IMO.